### PR TITLE
add d implementation

### PR DESCRIPTION
**Pull request by Moth-Tolias**
as mentioned on mastodon.

- [x] I own the full rights to my contribution, and agree to release it under the terms of the Creative Commons Zero Public Domain Dedication

### CODE REVIEW

**Review by exaskye — COMMENTED** [review]
Looks OK to me, but I don't know D. @emilyploszaj knows D and offered to do a review. I'll approve if she does.

File: d/source/flexver.d [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+import std.ascii: isDigit;
+import std.uni: byCodePoint;
+
+///
+int compare(in string first, in string second) pure @safe
+{
+	auto decomposedFirst = decompose(first);
+	auto decomposedSecond = decompose(second);
+
+	int componentIndex;
```
Comment (emilyploszaj) [inline d/source/flexver.d:13]: This could be an index in the foreach instead of manually tracking

File: d/source/flexver.d [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+		break;
+
+	case ComponentType.None:
+		return false;
+	}
+
+	return true;
+}
+
+version (unittest)
```
Comment (emilyploszaj) [inline d/source/flexver.d]: Why have unit test utils in a separate block instead of inside of the unittest block itself?

**Review by emilyploszaj — COMMENTED** [review]
Looks alright except for a couple crits, I can't verify the flexver logic itself but it's probably fine if tests are passing.

File: d/source/flexver.d (hunk 1 above)
Comment (Moth-Tolias) [reply d/source/flexver.d:13]: it's needed outside of the foreach, line 35

File: d/source/flexver.d (hunk 2 above)
Comment (Moth-Tolias) [reply d/source/flexver.d]: not sure. maybe we thought it was going to be used in multiple tests at one point?
we've moved them back inside.

**Review by emilyploszaj — APPROVED** [review]

**Review by exaskye — APPROVED** [review]

**Comment by exaskye** [comment]
Thanks!

**Pull request merged**